catroid/src/androidTest/java/org/catrobat/catroid/test/content/TranslationCommandsTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content;
+
+import android.content.res.Configuration;
+import android.test.AndroidTestCase;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.common.LookData;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.Translatable;
+import org.catrobat.catroid.content.bricks.BroadcastBrick;
+import org.catrobat.catroid.content.bricks.NoteBrick;
+import org.catrobat.catroid.content.commands.CommandFactory;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.FormulaElement;
+import org.catrobat.catroid.formulaeditor.UserList;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.io.StorageHandler;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.catrobat.catroid.utils.UtilFile;
+import org.catrobat.catroid.utils.Utils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class TranslationCommandsTest extends AndroidTestCase {
```

Comment:
AndroidTestCase has been deprecated, it would be nice to have new tests in Junit4 (there shouldnt be a TestRule required for this since it isnt run as activity test).
for this test conversion should be rather straight forward, remove the "extends AndrdoiTestCase" and make test class not inherited, add @RunsWith(AndroidJunit4.class), and add \@Before \@After \@Test  annotations.

if test crashed due to something like "cannot create Handler" or "cannot call looper"  add the following rule:
	@Rule
	public UiThreadTestRule uiThreadTestRule = new UiThreadTestRule();
and annotate the test with 	@UiThreadTest
this happens with tests that run some part of the project that do try to initialize a new handler, which crashes because with regular AndroidJunit4 tests the test, and everything it does, is not executed on the UIThread of the android device...